projects/js-packages/components/components/button/index.tsx

```diff
@@ -46,7 +46,7 @@ export const Button: React.FC< ButtonProps > = ( {
 	return (
 		<WPButton
 			target={ externalTarget }
-			variant={ isExternalLink ? 'link' : variant }
+			variant={ variant === 'external-link' ? 'link' : variant }
```

Comment:
I can't say this change seems to actually make the code better. We're now duplicating the condition from line 39 instead of using the variable assigned there.

I guess this is what people mean when they talk about coding for the tool rather than for developers reading the code.

Reply:
No, that's not the case. It's just that we are on an older version of TS which isn't that smart to automatically understand what's going on here. Should I update `typescript` to latest version? It shouldn't be a problem though.

Reply:
It might also be worth noting that we have `"typescript"` as a dependency in 7 workspaces.

<img width="395" alt="Screenshot 2022-05-13 at 6 12 45 PM" src="https://user-images.githubusercontent.com/18226415/168285534-18fba405-a0bc-47e6-8c00-b5a4cce817b3.png">

Is there a way to improve it and use it from a single place like root or from js-tools?

Reply:
We put eslint and prettier in js-tools because we generally want to run those once from the monorepo root (particularly during pre-commit) rather than for each project.

js-test-runner exists (and was probably a mistake) because mocha apparently needs a ton of configuration.

But we don't put babel or jest or webpack or postcss in at the monorepo level, since what exactly is needed and how it's configured seems specific to each project in the monorepo and we run those things from each project during build or unit testing. Typescript would seem to fall into the same category IMO.

You could go to [#23363](https://href.li?https://github.com/Automattic/jetpack/issues/23363) and check the box by "Update dependency typescript" to have Renovate create a PR for updating it everywhere in the monorepo.

Reply:
> But we don't put babel or jest or webpack or postcss in at the monorepo level, since what exactly is needed and how it's configured seems specific to each project in the monorepo and we run those things from each project during build or unit testing. Typescript would seem to fall into the same category IMO.

Yes, I agree.


> You could go to [#23363](https://href.li?https://github.com/Automattic/jetpack/issues/23363) and check the box by "Update dependency typescript" to have Renovate create a PR for updating it everywhere in the monorepo.

Done. Thanks

Reply:
I have reverted the change because we now have a smarter version of TS.